analyzers/tests/SonarAnalyzer.UnitTest/Rules/StaticFieldWrittenFromInstanceMemberTest.cs

```diff
@@ -18,28 +18,67 @@
  * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
  */
 
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CSharp;
 using SonarAnalyzer.Rules.CSharp;
 
 namespace SonarAnalyzer.UnitTest.Rules
 {
     [TestClass]
     public class StaticFieldWrittenFromInstanceMemberTest
     {
+        private readonly VerifierBuilder<StaticFieldWrittenFromInstanceMember> builder = new();
+
         [TestMethod]
         public void StaticFieldWrittenFromInstanceMember() =>
-            OldVerifier.VerifyAnalyzer(@"TestCases\StaticFieldWrittenFromInstanceMember.cs",
-                                    new StaticFieldWrittenFromInstanceMember(),
-                                    ParseOptionsHelper.FromCSharp8,
-                                    MetadataReferenceFacade.NETStandard21);
+            builder.AddPaths(@"StaticFieldWrittenFromInstanceMember.cs").WithOptions(ParseOptionsHelper.FromCSharp8).AddReferences(MetadataReferenceFacade.NETStandard21).Verify();
+
+        [TestMethod]
+        public async Task SecondaryIssueInReferencedCompilation()
+        {
+            const string firstClass =
+                @"
+public class Foo
+{
+    public static int Count = 0; // Secondary
+}
+";
+
+            const string secondClass =
+                @"
+public class Bar
+{
+    public int Increment() => Foo.Count++;
+}
+";
+
+            var analyzers = ImmutableArray<DiagnosticAnalyzer>.Empty.Add(new StaticFieldWrittenFromInstanceMember());
+            var firstCompilation = CreateCompilation(CSharpSyntaxTree.ParseText(firstClass), "First").WithAnalyzers(analyzers).Compilation;
+            var secondCompilation = CreateCompilation(CSharpSyntaxTree.ParseText(secondClass), "Second")
+                                    .AddReferences(firstCompilation.ToMetadataReference())
+                                    .WithAnalyzers(analyzers);
+
+            firstCompilation.GetDiagnostics().Should().BeEmpty();
+
+            var result = await secondCompilation.GetAnalyzerDiagnosticsAsync();
```

Comment:
Assertions should be kept together (ArrangeActAssert),
```suggestion
            var result = await secondCompilation.GetAnalyzerDiagnosticsAsync();

            firstCompilation.GetDiagnostics().Should().BeEmpty();
```

```